### PR TITLE
feat(LERN-84): implemented teacher dashboard

### PR DESCRIPTION
**Pull request by kieranslost**

### CODE REVIEW

File: frontend/src/state/api/teachers.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from '@tanstack/react-query';
+import { TeacherResourceService } from './generated';
+
+export const useTeachersQuery = (schoolId: string) => {
```
Comment (deepspace1000) [inline frontend/src/state/api/teachers.ts:4]:
```suggestion
export const useGetTeachersForSchoolQuery = (schoolId: string) => {
```
`useTeachersQuery` is a bit to generic

File: frontend/src/pages/TeacherPage.tsx [hunk 2]
```diff
@@ -1,12 +1,28 @@
-import { useUserQuery } from '../state/api/user.ts';
+import { Box, Button, Grid, Typography } from '@mui/joy';
+import { TeacherTable } from '../components/TeacherTable.tsx';
 
 export function TeacherPage() {
-  const backendDto = useUserQuery();
-
   return (
     <>
```
Comment (deepspace1000) [inline frontend/src/pages/TeacherPage.tsx:6]: not needed

File: frontend/src/components/TeacherTable.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { whoamiAtom } from '../state/api/whoami.ts';
+
+export function TeacherTable() {
+  const [{ data: whoami }] = useAtom(whoamiAtom);
+  const teachers = useTeachersQuery(whoami.schools[0].id);
+
+  return (
+    <>
+      <div style={{ marginTop: '40px' }}>
+        <Sheet sx={{ height: 500, overflow: 'auto', backgroundColor: '#f6fbfa' }}>
```
Comment (deepspace1000) [inline frontend/src/components/TeacherTable.tsx:14]: I think you don't need the sheet here. you can use the div for all this stuff.

File: frontend/src/pages/TeacherPage.tsx [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
-      <h1>Lehrer 📚🥳</h1>
-      <div>{backendDto}</div>
+      <Box
+        sx={{
+          width: '1320px',
+          margin: '40px auto',
+
+          padding: '20px',
+        }}>
+        <Grid container spacing={2} sx={{ flexGrow: 1 }}>
```
Comment (deepspace1000) [inline frontend/src/pages/TeacherPage.tsx:14]: i would still go with css grid 🙃

File: frontend/src/pages/TeacherPage.tsx [hunk 5]
```diff
@@ -1,12 +1,28 @@
...
+          <Grid xs={6} margin={0}>
```
Comment (deepspace1000) [inline frontend/src/pages/TeacherPage.tsx:15]: here too

File: frontend/src/pages/TeacherPage.tsx [hunk 6]
```diff
@@ -1,12 +1,28 @@
...
+import { TeacherTable } from '../components/TeacherTable.tsx';
 
 export function TeacherPage() {
-  const backendDto = useUserQuery();
-
   return (
     <>
-      <h1>Lehrer 📚🥳</h1>
-      <div>{backendDto}</div>
+      <Box
```
Comment (deepspace1000) [inline frontend/src/pages/TeacherPage.tsx:7]: You could create a component `ContentSection` which would have this fixed with and so you can reuse it for every page.

File: frontend/src/pages/TeacherPage.tsx [hunk 7]
```diff
@@ -1,12 +1,28 @@
...
+        sx={{
+          width: '1320px',
+          margin: '40px auto',
+
+          padding: '20px',
+        }}>
+        <Grid container spacing={2} sx={{ flexGrow: 1 }}>
+          <Grid xs={6} margin={0}>
+            <Typography level="h3" component={'h2'} margin={0}>
```
Comment (deepspace1000) [inline frontend/src/pages/TeacherPage.tsx:16]:
```suggestion
            <Typography level="h3" margin={0}>
```

**Review by deepspace1000 — CHANGES_REQUESTED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

File: frontend/src/pages/TeacherPage.tsx [hunk 8]
```diff
@@ -1,28 +1,15 @@
...
-            <Button>Lehrer hinzufügen</Button>
-          </Grid>
-        </Grid>
-        <TeacherTable />
-      </Box>
-    </>
+    <ContentSection>
+      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
+        <Typography level="h3">Lehrerübersicht</Typography>
+        <Button style={{ float: 'right' }}>Lehrer hinzufügen</Button>
```
Comment (deepspace1000) [inline frontend/src/pages/TeacherPage.tsx:10]: could you use something like `alignSelf: right` or `alignSelf: flexEnd` ?

File: frontend/src/components/TeacherTable.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  );
+}
+
+const TableHead = styled('th')`
+  background-color: #f6fbfa !important;
+  border-bottom: 2px solid #5ec2b7 !important;
+  width: 20%;
+`;
+
+const TableBody = styled('td')`
```
Comment (deepspace1000) [inline frontend/src/components/TeacherTable.tsx:45]:
```suggestion
const TableData = styled('td')`
```

Comment (deepspace1000) [inline frontend/src/components/TeacherTable.tsx:45]:
```suggestion
const TableData = styled('td')`
```

Comment (deepspace1000) [inline frontend/src/components/TeacherTable.tsx:45]:
```suggestion
const TableData = styled('td')`
```

**Review by deepspace1000 — COMMENTED** [review]
Already looks a lot better now 😃

File: frontend/src/pages/TeacherPage.tsx (hunk 8 above)
Comment (kieranslost) [reply frontend/src/pages/TeacherPage.tsx:10]: turns out since i used "space-between" i dont need this code anyways

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

File: frontend/src/pages/TeacherPage.tsx (hunk 8 above)
Comment (deepspace1000) [reply frontend/src/pages/TeacherPage.tsx]: Well thats even better 😄

**Review by deepspace1000 — COMMENTED** [review]
Nice now fix the sonar issue und you can merge

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Review by deepspace1000 — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Pull request merged**